### PR TITLE
New Entry: C++ Templates

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

From target keyword list.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/cpp/concepts/templates/templates.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  - 'Methods'
+  - 'Functions'
+  - 'OOP'
+  - 'Parameters'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+**Templates** provide the ability to use a [type](https://www.codecademy.com/resources/docs/cpp/data-types) as a parameter in [functions](https://www.codecademy.com/resources/docs/cpp/functions) and [classes](https://www.codecademy.com/resources/docs/cpp/classes). Such classes and functions are referred to as using a generic type. This provides the ability to define a set of related classes or functions that can operate on many different types with a single declaration.
```
Comment (Dusch4593) [inline content/cpp/concepts/templates/templates.md:21]:
```suggestion
**Templates** provide the ability to use a [data type](https://www.codecademy.com/resources/docs/cpp/data-types) as a parameter in [functions](https://www.codecademy.com/resources/docs/cpp/functions) and [classes](https://www.codecademy.com/resources/docs/cpp/classes). These are referred to as generic types. This provides the ability to define a set of related classes or functions that can operate on many different types with a single declaration.
```

File: content/cpp/concepts/templates/templates.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Classes'
+  - 'Objects'
+  - 'Data Types'
+  - 'Constructors'
+  - 'Methods'
+  - 'Functions'
+  - 'OOP'
+  - 'Parameters'
```
Comment (Dusch4593) [inline content/cpp/concepts/templates/templates.md:15]:
```suggestion
  - 'Parameters'
  - 'Templates'
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Just a few suggestion from me for this one 😄

**Comment by SSwiniarski** [comment]
@Dusch4593, implemented your suggestions.

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski Looks good for the next review 😄

**Review by KTom101 — COMMENTED** [review]
@SSwiniarski Just a few edits from me, mainly structural.

**Comment by SSwiniarski** [comment]
@KTom101 I implemented your suggestions

**Review by KTom101 — APPROVED** [review]
@SSwiniarski Looks good!

**Comment by KTom101** [comment]
@Dusch4593 Was there anything you wanted to add before merging?

**Comment by Dusch4593** [comment]
@KTom101 LGTM! 👍🏻 I just linked to out to the [operators entry](https://www.codecademy.com/resources/docs/cpp/operators) in b68599d355cb1c09e4c7a6e1f4d557600b888a9c.

**Pull request merged**